### PR TITLE
Back to top button

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

File: src/back-to-top/index.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    this.handleScroll = this.handleScroll.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll, true);
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleOnScroll);
```
Comment (joefitter) [inline src/back-to-top/index.jsx]:
```suggestion
    window.removeEventListener('scroll', this.handleScroll);
```

File: src/back-to-top/index.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  handleScroll() {
+    if (!this.state.isVisible && window.pageYOffset > this.props.showAt) {
+      this.setState({ isVisible: true });
+    } else if (this.state.isVisible && window.pageYOffset < this.props.showAt) {
+      this.setState({ isVisible: false });
+    }
+  }
+
+  scrollToTop() {
+    window.scrollTo(0, 0);
```
Comment (joefitter) [inline src/back-to-top/index.jsx]: this is an instant jump - you can pass it behaviour: smooth for a nice animated scroll
```js
window.scrollTo({
  top: 0,
  left: 0,
  behavior: 'smooth'
});
```

File: src/wrapper/index.jsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import React, { Fragment } from 'react';
-import { Notification } from '../';
+import { BackToTop, Notification } from '../';
 
 const Wrapper = ({ children }) => (
   <Fragment>
     <Notification />
     { children }
+    <BackToTop showAt={400} />
```
Comment (joefitter) [inline src/wrapper/index.jsx]: as you're just using the default I'd probably remove the prop here

File: src/back-to-top/index.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.back-to-top {
+  background: $white;
+  padding: 0 10px;
+
+  position: sticky;
```
Comment (lennym) [inline src/back-to-top/index.scss]: This is _quite_ badly supported cross browser. Have you tested in any not-latest-Chrome browsers?

https://caniuse.com/#feat=css-sticky

**Pull request merged**